lib/pbkit/pbkit.c

```diff
@@ -2544,9 +2546,22 @@ void pb_fill(int x, int y, int w, int h, DWORD color)
     y1=y;
     x2=x+w;
     y2=y+h;
-    
-    //if you supply 32 bits color and res is 16 bits, apply function below
-    //color=((color>>8)&0xF800)|((color>>5)&0x07E0)|((color>>3)&0x001F);
+
+    switch(pb_ColorFmt) {
+    case NV097_SET_SURFACE_FORMAT_COLOR_LE_X1R5G5B5_Z1R5G5B5:
+    case NV097_SET_SURFACE_FORMAT_COLOR_LE_X1R5G5B5_O1R5G5B5:
+        color=((color>>16)&0x8000)|((color>>7)&0x7C00)|((color>>5)&0x03F0)|((color>>3)&0x001F);
+        break;
+    case NV097_SET_SURFACE_FORMAT_COLOR_LE_R5G6B5:
+        color=((color>>8)&0xF800)|((color>>5)&0x07E0)|((color>>3)&0x001F);
+        break;
+    case NV097_SET_SURFACE_FORMAT_COLOR_LE_A8R8G8B8:
+        // Nothing to do, input is A8R8G8B8
+        break;
+    default:
+        assert(false);
```

Comment:
This should also throw an error message which explains the problem.

More modes can be added in the future.

Reply:
`assert` is ok for now I think